### PR TITLE
kubeadm: Remove the usage to print the default component configs for `reset` and `join`

### PR DESCRIPTION
**Pull request by chendave**
component configs is only needed for `kubeadm init`, the `join` and `reset` doesn't need to provid the config with component configs.

/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
kubeadm: print the default component configs for `reset` and `join` is now not supported
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by SataQiu** [comment]
> Even it's possible by the CLI, but it shouldn't be guaranteed by any testcases.

I'm wondering if we should disable the `--component-configs` flag for `reset`. 
What do you think? @neolit123 @pacoxu

**Comment by neolit123** [comment]
> > Even it's possible by the CLI, but it shouldn't be guaranteed by any testcases.
> 
> I'm wondering if we should disable the `--component-configs` flag for `reset`. What do you think? @neolit123 @pacoxu

print for join also doesn’t need it?
if we already have it for join, it needs deprecation.

instead...we can keep it enabled everywhere but a No-op for reset, join. i might be missing some context.

**Comment by SataQiu** [comment]
> we can keep it enabled everywhere but a No-op for reset, join.

Yes, these default configurations are only printed when the `--component-configs` flag is specified, it seems harmless.

**Review by neolit123 — COMMENTED** [review]
/approve
seems ok to clean these

**Comment by pacoxu** [comment]
> print for join also doesn’t need it? if we already have it for join, it needs deprecation.

I prefer to deprecate it as it is not needed(harmless as well).

> instead...we can keep it enabled everywhere but a No-op for reset, join. i might be missing some context.

I will make users confused like the below. 
```
kubeadm config print reset-defaults --component-configs=KubeProxyConfiguration
```

> I'm wondering if we should disable the `--component-configs` flag for `reset`.

+1

**Comment by chendave** [comment]
/hold

this needs to be polished, we can do more instead of just remove those testcases.

**Comment by chendave** [comment]
/retitle kubeadm: Deprecate the default values of component config for `reset` and `join`

/unhold

File: cmd/kubeadm/app/cmd/config.go [hunk 1]
```diff
@@ -115,6 +115,7 @@ func newCmdConfigPrintResetDefaults(out io.Writer) *cobra.Command {
 
 func newCmdConfigPrintActionDefaults(out io.Writer, action string, configBytesProc func() ([]byte, error)) *cobra.Command {
 	kinds := []string{}
+	deprecatedMsg := fmt.Sprintf("Print the default value of component config API objects is deprecated for action: %s, it will be ingored if set and will be removed in a future release.", action)
```
Comment (pacoxu) [inline cmd/kubeadm/app/cmd/config.go:118]: typo: ignore.

Comment (chendave) [reply cmd/kubeadm/app/cmd/config.go]: Done.

print it to stdout use `fmt.Fprintln` instead.

File: cmd/kubeadm/app/cmd/config.go [hunk 2]
```diff
@@ -115,6 +115,7 @@ func newCmdConfigPrintResetDefaults(out io.Writer) *cobra.Command {
 
 func newCmdConfigPrintActionDefaults(out io.Writer, action string, configBytesProc func() ([]byte, error)) *cobra.Command {
 	kinds := []string{}
+	deprecatedMsg := fmt.Sprintf("Print the default value of component config API objects is deprecated for action: %s, it will be ignored if set and will be removed in a future release.", action)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/config.go:118]: i may have misunderstood something around our discussion for UpgradeConfiguration?
we should continue printing the kubeadm's understanding of default component config, but we can stop managing component config via --config on upgrade.

**Review by neolit123 — COMMENTED** [review]
can we go without deprecation here and just remove the usage of component-configs for anything but init?

it feels like a bug to allow join, reset...etc to use these.

**Comment by chendave** [comment]
> can we go without deprecation here and just remove the usage of component-configs for anything but init?

this is following the suggestion from @pacoxu , your input here? are you okay if we go without deprecation?

From @pacoxu 
> I prefer to deprecate it as it is not needed(harmless as well).

**Comment by chendave** [comment]
also, this needs a release note, will add it with in the next iteration.

**Comment by pacoxu** [comment]
> it feels like a bug to allow join, reset...etc to use these.

Just remove it if this is a bug 😄.

**Review by neolit123 — COMMENTED** [review]
/approve
SGTM

please always prefix kubeadm release notes with `kubeadm: `

**Comment by chendave** [comment]
> please always prefix kubeadm release notes with kubeadm:

Done.

**Comment by chendave** [comment]
/hold

testcase need to be updated as well.

**Comment by chendave** [comment]
/unhold

updated testcases since the `component-configs` flags is now not available for `reset` and `join`.

**Pull request merged**